mw-webapp/src/logic/userPage/UserPage.tsx

```diff
@@ -60,22 +61,40 @@ export const UserPage = (props: UserPageProps) => {
   const isPageOwner = !!user && !!userPreview && user.uuid === userPreview.uuid;
 
   /**
-   * Load user
+   * Callback that is called to fetch data
    */
-  const getUser = async () => {
-    const userPreviewData = await UserPreviewDAL.getUserPreview(props.uuid);
+  const loadData = () => UserPreviewDAL.getUserPreview(props.uuid);
+
+  /**
+   * Callback that is called to validate data
+   */
+  const validateData = (data: UserPreview) => !!data.uuid;
+
+  /**
+   * Callback that is called on fetch and validation success
+   */
+  const onSuccess = (data: UserPreview) => {
+    setUserPreview(data);
+  };
+
+  /**
+   * Callback this is called on fetch or validation error
+   */
+  const onError = () => {
     // Navigate to PageError if transmitted user's uuid is not exist
-    if (!userPreviewData.uuid) {
-      navigate(pages.page404.getPath({}));
-    }
-    setUserPreview(userPreviewData);
+    navigate(pages.page404.getPath({}));
   };
 
-  useEffect(() => {
-    getUser();
-  }, [props.uuid]);
+  const {isLoading} = useLoad({
+    loadData,
+    validateData,
+    onSuccess,
+    onError,
+    dependency: props.uuid,
+  },
+  );
 
-  if (!userPreview) {
+  if (isLoading || !userPreview) {
```

Comment:
do we still need to check "!userPreview"?

Reply:
Just for TS, so userPreview is not undefined in return

Reply:
But if is loading finished and userPreview is not exist - this case mostly looks like a bug arequired to throw a error instead of showing loader?  It's not only about typescript as I see

Reply:
This is impossible because we have onError callback on this page, i don't know how to skip userPreview check for render other way

Reply:
Maked onError not optional

Reply:
But it looks a little strange semantically for my opinion. When I see this line I think that there are two possible conditions when data is in "loading" state. Should we check all states the same way? If yes - why do we need isLoading state?

Reply:
Removed isLoading from this page